### PR TITLE
Added GNN example and updated readme

### PR DESCRIPTION
**Pull request by wangxiaoyunNV**
Fixes # .

### Description
Add graphsage example, updated readme for GNN background and pyG

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [x] Documentation updated.

### CODE REVIEW

File: examples/advanced/gnn-pt/jobs/gnn-pt/app/custom/graphsage.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+# (4) test received model for model selection
+def encode(loader):
+    net.eval()
+    xs, ys = [], []
+    for data in loader:
+        data = data.to(device)
+        xs.append(net(data.x, data.edge_index).cpu())
+        ys.append(data.y.cpu())
+    return torch.cat(xs, dim=0), torch.cat(ys, dim=0)
+
```
[the remark this answers is not in the document]
Comment (YuanTingHsieh) [reply examples/advanced/gnn-pt/jobs/gnn-pt/app/custom/graphsage.py:70]: For this one, L72 - L80 is doing this

File: examples/advanced/gnn-pt/jobs/gnn-pt/app/custom/graphsage.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+# (1) initializes NVFlare client API
+flare.init()
+# (2) gets FLModel from NVFlare
+input_model = flare.receive()
+
+# (3) loads model from NVFlare
+net.load_state_dict(input_model.params)
+net.to(device)
+
+# (4) test received model for model selection
```
[the remark this answers is not in the document]
Comment (YuanTingHsieh) [reply examples/advanced/gnn-pt/jobs/gnn-pt/app/custom/graphsage.py:61]: The whole L62 - L80 is doing this, so I leave the comments here, that is for the whole section. but we can remove if we find it confusing.

File: examples/advanced/gnn-pt/jobs/gnn-pt/app/custom/graphsage.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+with torch.no_grad():
+    x, y = encode(train_loader)
+    clf = MultiOutputClassifier(SGDClassifier(loss="log", penalty="l2"))
+    clf.fit(x, y)
+    train_f1 = f1_score(y, clf.predict(x), average="micro")
+
+    # Evaluate on test set:
+    x, y = encode(test_loader)
+    test_f1 = f1_score(y, clf.predict(x), average="micro")
```
Comment (YuanTingHsieh) [inline examples/advanced/gnn-pt/jobs/gnn-pt/app/custom/graphsage.py:80]: this is global, if we want we call it "global_test_f1" as Chester suggested.

File: examples/advanced/gnn-pt/README.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+Example of using [NVIDIA FLARE](https://nvflare.readthedocs.io/en/main/index.html) to train an image classifier
+using federated averaging ([FedAvg](https://arxiv.org/abs/1602.05629))
+and [PyTorch](https://pytorch.org/) as the deep learning training framework.
+
+
+
+### Background of Graph Neural Network
+
+Graph Neural Networks (GNNs) are a promising research topic and industry market, with potential applications in various domains, including social networks, e-commerce, recommendation systems, and more.
```
[the remark this answers is not in the document]
Comment (wangxiaoyunNV) [reply examples/advanced/gnn-pt/README.md:11]: we should use "are", because GNNs refer to multiple kinds of graph neural networks. I rewrote this sentence.

File: examples/advanced/gnn-pt/README.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+GNNs excel in learning, modeling, and leveraging complex relationships within graph-structured data. They combine local and global information, incorporate structural knowledge, adapt to diverse tasks, handle heterogeneous data, support transfer learning, scale for large graphs, offer interpretable insights, and achieve impressive performance. 
+[GraphSAGE](https://arxiv.org/pdf/1706.02216.pdf) is a widely used GNN due to its ability to perform inductive learning on graph-structured data and scalability. 
+Here we give an example of classifying protein roles based on their cellular functions from gene ontology with unsupervised [GraphSAGE](https://arxiv.org/pdf/1706.02216.pdf). The dataset we are using is
```
[the remark this answers is not in the document]
Comment (wangxiaoyunNV) [reply examples/advanced/gnn-pt/README.md:14]: graphSAGE has supervised and unsupervised setting. So may I should add "setting" after graphsage

File: examples/advanced/gnn-pt/jobs/gnn-pt/app/custom/graphsage.py [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+        loss = F.binary_cross_entropy_with_logits(link_pred, data.edge_label)
+        loss.backward()
+        optimizer.step()
+
+        total_loss += float(loss) * link_pred.numel()
+        total_examples += link_pred.numel()
+
+
+# (5) construct the FLModel to returned back
+output_model = flare.FLModel(params=net.cpu().state_dict(), params_type="FULL", metrics={"f1": global_test_f1})
```
Comment (YuanTingHsieh) [inline examples/advanced/gnn-pt/jobs/gnn-pt/app/custom/graphsage.py:105]: Let's change this line to be:

`output_model = flare.FLModel(params=net.cpu().state_dict(), params_type="FULL", metrics={"test_f1": global_test_f1})`

File: examples/advanced/gnn-pt/jobs/gnn-pt/app/config/config_fed_server.json [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+      "path": "nvflare.app_common.aggregators.intime_accumulate_model_aggregator.InTimeAccumulateWeightedAggregator",
+      "args": {
+        "expected_data_kind": "WEIGHTS"
+      }
+    },
+    {
+      "id": "model_selector",
+      "name": "IntimeModelSelector",
+      "args": {
+        "key_metric": "test_f1"
```
Comment (YuanTingHsieh) [inline examples/advanced/gnn-pt/jobs/gnn-pt/app/config/config_fed_server.json:38]: This key_metric should match the key of the metric in the returned FLModel.
That's why I ask to change the line over there to "test_f1"

File: examples/advanced/gnn-pt/nohup.out [hunk 8]
```diff
@@ -0,0 +1,64 @@
+2023-09-06 20:23:59,182 - SimulatorRunner - INFO - Create the Simulator Server.
```
Comment (YuanTingHsieh) [inline examples/advanced/gnn-pt/nohup.out:1]: Let's not include the output log

File: examples/advanced/gnn-pt/README.md [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+
+
+### 1. Install NVIDIA FLARE
+
+Follow the [Installation](https://nvflare.readthedocs.io/en/main/quickstart.html) instructions.
+Install additional requirements:
+
+```
+pip3 install -r requirements.txt
+```
```
[the remark this answers is not in the document]
Comment (wangxiaoyunNV) [reply examples/advanced/gnn-pt/README.md:30]: currently, it is the same data shared by all clients. 
If we want to split the data, we can let each client have a number of connected components.
